integration/hive/src/main/java/org/apache/carbondata/hive/CarbonDictionaryDecodeReadSupport.java

```diff
@@ -228,9 +229,13 @@ private Writable createWritablePrimitive(Object obj, CarbonColumn carbonColumn)
       case LONG:
         return new LongWritable((long) obj);
       case SHORT:
-        return new ShortWritable((Short) obj);
+        return new ShortWritable((short) obj);
       case DATE:
-        return new DateWritable(new Date(((Integer) obj).longValue()));
+        Calendar c = Calendar.getInstance();
+        c.setTime(new Date(0));
```

Comment:
Does it mean current time? and then reset the year to current year?

Reply:
Yup

Reply:
There are changes in master code for date so these changes are required for getting the right values at hive side